src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy

```diff
@@ -68,6 +68,13 @@ class ReportSinceInstant implements Runnable {
               .map(Result::getValue)
               .peek(it -> log.info("\tUpdating $it"))
               .forEach(statusReporter::reportSampleStatusUpdate)
+      def errors = updatedSamples.stream()
+              .filter(Result::isError)
+              .map(Result::getError).collect()
+      if (errors.size() > 0) {
+        log.warn("Encountered ${errors.size()} errors.")
+        errors.forEach((Exception e) -> log.error(e.getMessage(), e))
+      }
```

Comment:
I wonder if we should pass errors upstream in the presence of errors. Think of this use case: you have a collection of queried samples from the LIMS, lets say 5, and 2 of them cannot be reported to the sample tracking service. What is your strategy then? Updating the time instance would meen, that we lost the sample update information for these 2 samples.

We need a recovery strategy imo.

Reply:
Good point. We could change it so that there is a check for any errors first and we exit directly without going the path of successful updates. What do you think?